java/src/Ice/src/main/java/com/zeroc/Ice/ObjectAdapter.java

```diff
@@ -1339,13 +1337,13 @@ private ObjectPrx newProxy(Identity ident, String facet) {
   private ObjectPrx newDirectProxy(Identity ident, String facet) {
     // Create a reference and return a proxy for this reference.
     var ref = _instance.referenceFactory().create(ident, facet, _reference, _publishedEndpoints);
-    return _instance.proxyFactory().referenceToProxy(ref);
+    return (ref == null) ? null : new com.zeroc.Ice._ObjectPrxI(ref);
```

Comment:
The null check is not required. The caller already called `checkIdentity`.

Reply:
Fixed.